Spokestack/NLUTensorflowTokenizer.swift

```diff
@@ -38,23 +38,25 @@ internal struct BertTokenizer {
         var normalizedTokens: [String] = []
         var indices: [Int] = []
         var encoded: [Int] = []
-        // then componentize and wordpiece the whitespaced tokens, creating an even bigger set of normalized tokens
+        // then componentize and wordpiece the whitespaced tokens, creating an even bigger list of normalized tokens
         for (id, wt) in whitespacedTokens.enumerated() {
-            let tokens
-                = self
+            let tokens = self
                 .componentize(wt)
                 .flatMap({ self.wordpiece(text: $0) })
             normalizedTokens += tokens
             // finally encode the normalized tokens
             for t in tokens {
-                encoded.append(self.encodings[t] ?? -1)
+                if let code = self.encodings[t] {
+                    encoded.append(code)
+                } else if let unk = self.encodings[unknownPiece] {
+                    encoded.append(unk)
+                } else {
+                    throw NLUError.tokenizer("NLU tokenizer could not encode the tokenized text.")
```

Comment:
Not having an encoding for `[UNK]` is really due to a bad vocab file, and that's probably better checked for at initialization time rather than the first time you happen to hit one in input text.